pkg/objectstate/sched/sched.go

```diff
@@ -0,0 +1,50 @@
+package sched
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nrsv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+)
+
+const DefaultSchedulerPluginImage = "quay.io/openshift-kni/scheduler-plugins:4.10-snapshot"
+
+func UpdateDeploymentImageSettings(dp *appsv1.Deployment, userImageSpec, builtinImageSpec string, builtinPullPolicy corev1.PullPolicy) {
+	// There is only a single container
+	cnt := &dp.Spec.Template.Spec.Containers[0]
+	if userImageSpec != "" {
+		// we don't really know what's out there, so we minimize the changes.
+		cnt.Image = userImageSpec
+		klog.V(3).InfoS("Exporter image", "reason", "user-provided", "pullSpec", userImageSpec)
+		return
+	}
+
+	if builtinImageSpec == "" {
+		cnt.Image = DefaultSchedulerPluginImage
+		klog.V(3).InfoS("Exporter image", "reason", "default", "pullSpec", builtinImageSpec)
+		return
+	}
+
+	cnt.Image = builtinImageSpec
+	cnt.ImagePullPolicy = builtinPullPolicy
+	klog.V(3).InfoS("Exporter image", "reason", "builtin", "pullSpec", builtinImageSpec, "pullPolicy", builtinPullPolicy)
+}
+
+func UpdateDeploymentConfigMapSettings(dp *appsv1.Deployment, cmName string) {
+	for i := range dp.Spec.Template.Spec.Volumes {
+		if dp.Spec.Template.Spec.Volumes[i].Name == "etckubernetes" {
```

Comment:
let's make this a constant, or let's just replace the `Volume` definition entirely:
```
func UpdateDeploymentConfigMap(dp *appsv1.Deployment, configMapName string) {
	spec := &dp.Spec.Template.Spec // shortcut
     // ...

	spec.Volumes[0] = newSchedConfigVolume(schedVolumeConfigName, configMapName)
}

func newSchedConfigVolume(schedVolumeConfigName, configMapName string) corev1.Volume {
	return corev1.Volume{
		Name: schedVolumeConfigName,
		VolumeSource: corev1.VolumeSource{
			ConfigMap: &corev1.ConfigMapVolumeSource{
				LocalObjectReference: corev1.LocalObjectReference{
					Name: configMapName,
				},
			},
		},
	}
}

Reply:
this still seems relevant